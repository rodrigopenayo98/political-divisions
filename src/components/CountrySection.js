import PropTypes from 'prop-types';
import React from 'react';

function CountrySection({ countries }) {
  if (!countries || countries.length === 0) {
    return (
      <p className="">
        No countries available.
      </p>
    );
  }

  return (
    <div>
      <ul>
        {countries.map((country) => (
          <li key={country.countryCode}>
            <p>
              Image:
              <img src={country.img} alt={country.countryName} />
            </p>
            <h3>{country.countryName}</h3>
            <p>
              Currency:
              {country.currencyCode}
            </p>
            <p>
              Area:
              {country.areaInSqKm}
              {' '}
              sq km
            </p>
            <p>
              Continent:
              {country.continentName}
            </p>
            <p>
              Population:
              {country.population}
            </p>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default CountrySection;

CountrySection.propTypes = {
  countries: PropTypes.arrayOf(PropTypes.shape({
    countryCode: PropTypes.string.isRequired,
    countryName: PropTypes.string.isRequired,
    currencyCode: PropTypes.string.isRequired,
    areaInSqKm: PropTypes.number.isRequired,
    continentName: PropTypes.string.isRequired,
    population: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
  })).isRequired,
};
