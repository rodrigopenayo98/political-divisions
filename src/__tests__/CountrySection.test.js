import React from 'react';
import renderer from 'react-test-renderer';
import CountrySection from '../components/CountrySection';

describe('CountrySection component', () => {
  it('renders correctly with a geonameId', () => {
    const component = renderer.create(<CountrySection geonameId="12345" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
