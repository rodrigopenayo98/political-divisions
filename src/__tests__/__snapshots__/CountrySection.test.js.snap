// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySection component renders correctly with a geonameId 1`] = `
<div>
  <div>
    <p
      className="span2"
    >
      Loading...
    </p>
  </div>
</div>
`;
